### PR TITLE
opts: Add SigningRoot, HttpsRoots loader

### PR DESCRIPTION
**Pull request by ChriMarMe**

### CODE REVIEW

**Comment by ChriMarMe** [comment]
Not quite done yet. Creating test data and writing tests.

File: opts/certs_test.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package opts
+
+import (
+	"errors"
+	"testing"
+
+	"golang.org/x/sys/unix"
+)
+
+func TestDecodePem(t *testing.T) {
```
Comment (jensdrenhaus) [inline opts/certs_test.go:10]: What about MultiCert pem files? Test this, too, or is there no effect?

File: opts/certs.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/system-transparency/stboot/stlog"
+)
+
+type SigningRoot struct {
```
Comment (jensdrenhaus) [inline opts/certs.go:12]: Why is this struct necessary?

File: opts/certs.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	stlog.Debug("    Common Name: %s", certs[0].Subject.CommonName)
+	stlog.Debug("  Valid from: %s", certs[0].NotBefore.String())
+	stlog.Debug("  Valid until: %s", certs[0].NotAfter.String())
+	o.SigningRoot = SigningRoot{certs[0]}
+	return nil
+}
+
+type HttpsRoots struct {
+	Certs []*x509.Certificate
+}
```
Comment (jensdrenhaus) [inline opts/certs.go:49]: Why is this struct necessary?

File: opts/opts.go [hunk 4]
```diff
@@ -22,6 +22,8 @@ type Opts struct {
 	Version int
 	Security
 	HostCfg
+	SigningRoot
+	HttpsRoots
```
Comment (jensdrenhaus) [inline opts/opts.go:26]: What about
```
SigningRoot *x509.Certificate
HttpsRoots []*x509.Certificate
```
Help me, if I miss something here, pls.

**Comment by jensdrenhaus** [comment]
I wonder if `type SigningRootFile struct` and `type HTTPSRootsFile struct` can be suashed into something like `type CertFile struct` or `type PEMFile struct`? 
I know one is one cert, the other is an array. But maybe there is a way to handle that. eg. check later during validation of opts.

File: opts/opts.go (hunk 4 above)
Comment (ChriMarMe) [reply opts/opts.go:26]: That was leftover from a different approach. Must have slipped

File: opts/certs_test.go (hunk 1 above)
Comment (ChriMarMe) [reply opts/certs_test.go:10]: Not even the qemu test runs with multi cert file :P

**Review by jensdrenhaus — APPROVED** [review]

**Pull request merged**